src/main/java/com/autotune/utils/TrialHelpers.java

```diff
@@ -61,195 +61,10 @@
-                configArrayObjects.put(envObject);
-
-                /*
-                 * metrics object
-                 * experimentTrialJSON -> deployments -> metrics
-                 */
-                JSONArray containerMetricArrayObjects = new JSONArray();
-                for (String metricName : podContainer.getContainerMetrics().keySet()) {
-                    Metric metricObjects = podContainer.getContainerMetrics().get(metricName);
-                    JSONObject obj = new JSONObject();
-                    obj.put(NAME, metricObjects.getName());
-                    obj.put(QUERY, metricObjects.getQuery());
-                    obj.put(DATASOURCE, metricObjects.getDatasource());
-                    containerMetricArrayObjects.put(obj);
-                }
-
-                /*
-                 * config object
-                 * experimentTrialJSON -> deployments -> config
-                 */
-                JSONObject configObject = new JSONObject();
-                configObject.put(IMAGE_NAME, podContainer.getStackName());
-                configObject.put(CONTAINER_NAME, podContainer.getContainerName());
-                configObject.put(CONFIG, configArrayObjects);
-                configObject.put(CONTAINER_METRICS, containerMetricArrayObjects);
-
-                containersJsonArray.put(configObject);
-            }
-
-            /*
-             * metrics object
-             * experimentTrialJSON -> deployments -> metrics
-             */
-            JSONArray podMetricArrayObjects = new JSONArray();
-            for (String metricName : deployment.getPodMetrics().keySet()) {
-                Metric metricObjects = deployment.getPodMetrics().get(metricName);
-                JSONObject obj = new JSONObject();
-                obj.put(NAME, metricObjects.getName());
-                obj.put(QUERY, metricObjects.getQuery());
-                obj.put(DATASOURCE, metricObjects.getDatasource());
-                podMetricArrayObjects.put(obj);
-            }
-
-            JSONObject deploymentObject = new JSONObject();
-            deploymentObject.put(TYPE, deployment.getDeploymentType());
-            deploymentObject.put(DEPLOYMENT_NAME, deployment.getDeploymentName());
-            deploymentObject.put(NAMESPACE, deployment.getDeploymentNameSpace());
-            deploymentObject.put(POD_METRICS, podMetricArrayObjects);
-            deploymentObject.put(CONTAINERS, containersJsonArray);
-
-            // Add this deployment tracker object to the deployment object array
-            deploymentsArrayObjs.put(deploymentObject);
-        }
-        experimentTrialJSON.put(DEPLOYMENTS, deploymentsArrayObjs);
-
-        return experimentTrialJSON;
+    public static String experimentTrialToJSON(ExperimentTrial experimentTrial) {
+        Gson gsonObj = new GsonBuilder().disableHtmlEscaping().setPrettyPrinting().create();
+        String gsonStr = gsonObj.toJson(experimentTrial);
+        return gsonStr;
```

Comment:
Nearly 200 lines replaced with 4, nice! (Ofcourse with the overhead of an additional package)

Reply:
Yes and I also found Gson pkg size in Kb's